packages/vuetify/src/components/VDataTable/VDataTable.ts

```diff
@@ -88,8 +95,8 @@ export default VDataIterator.extend({
     },
     customFilter: {
       type: Function,
-      default: defaultFilter,
-    } as PropValidator<typeof defaultFilter>,
+      default: customFilter,
+    } as PropValidator<typeof customFilter>,
```

Comment:
Use `PropType` instead of `PropValidator`, create a new `DataTableCustomFilterFunction` and expose it in vuetify/types/index.d.ts